src/pybind/mgr/dashboard/frontend/src/app/core/navigation/navigation/navigation.component.ts

```diff
@@ -78,12 +80,41 @@ export class NavigationComponent implements OnInit, OnDestroy {
         this.showTopNotification('motdNotificationEnabled', _.isPlainObject(motd));
       })
     );
+    this.router.events.subscribe((event: Event) => {
+      if (event instanceof NavigationEnd) {
+        this.setDisplayedSubMenu();
+      }
+    });
   }
 
   ngOnDestroy(): void {
     this.subs.unsubscribe();
   }
 
+  setDisplayedSubMenu(): void {
+    if (
+      this.router.url.startsWith('/hosts') || this.router.url.startsWith('/inventory')
+      || this.router.url.startsWith('/monitor') || this.router.url.startsWith('/services')
+      || this.router.url.startsWith('/osd') || this.router.url.startsWith('/configuration')
+      || this.router.url.startsWith('/crush-map') || this.router.url.startsWith('/mgr-modules')
+      || this.router.url.startsWith('/logs') || this.router.url.startsWith('/monitoring')
+    ) {
+      this.displayedSubMenu = 'cluster';
+    }
+    if (
+      this.router.url.startsWith('/block/rbd/') || this.router.url.startsWith('/block/mirroring')
+      || this.router.url.startsWith('/block/iscsi')
+    ) {
```

Comment:
```suggestion
    if (this.router.url.startsWith('/block/')) {
```

Reply:
@nizamial09 the routes inside Cluster menu doesn't start with `/cluster/`.
I am thinking of doing this.
```js
let pages = ['/hosts', '/inventory', '/monitor', '/services', '/osd' ... ];
let res = pages.some((page) => this.router.url.startsWith(page));
if (res)  this.displayedSubMenu = 'cluster';
```